packages/beacon-node/src/chain/chain.ts

```diff
@@ -638,135 +637,104 @@ export class BeaconChain implements IBeaconChain {
     return data && {block: data, executionOptimistic: false, finalized: true};
   }
 
-  async produceCommonBlockBody(blockAttributes: BlockAttributes): Promise<CommonBlockBody> {
-    const {slot, parentBlockRoot} = blockAttributes;
-    const state = await this.regen.getBlockSlotState(
-      toRootHex(parentBlockRoot),
-      slot,
-      {dontTransferCache: true},
-      RegenCaller.produceBlock
-    );
-
+  async produceCommonBlockBody(
+    blockAttributes: BlockAttributes & {currentState: CachedBeaconStateAllForks}
+  ): Promise<CommonBlockBody> {
     // TODO: To avoid breaking changes for metric define this attribute
     const blockType = BlockType.Full;
 
-    return produceCommonBlockBody.call(this, blockType, state, {
+    return produceCommonBlockBody.call(this, blockType, blockAttributes.currentState, {
       ...blockAttributes,
-      parentSlot: slot - 1,
+      parentSlot: blockAttributes.slot - 1,
     });
   }
 
-  produceBlock(blockAttributes: BlockAttributes & {commonBlockBody?: CommonBlockBody}): Promise<{
-    block: BeaconBlock;
-    executionPayloadValue: Wei;
-    consensusBlockValue: Wei;
-    shouldOverrideBuilder?: boolean;
-  }> {
-    return this.produceBlockWrapper<BlockType.Full>(BlockType.Full, blockAttributes);
+  async produceFullBlockBody(
```

Comment:
the name is misleading because we don't produce common block body after this method. How about `produceExecutionBlockBody`

also need to comment that this method does not produce common block body and we do that later, same to `produceBlindedBlockBody`